apps/studio/components/interfaces/Auth/ThirdPartyAuthForm/CreateAwsCognitoAuthDialog.tsx

```diff
@@ -0,0 +1,192 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Trash } from 'lucide-react'
+import { useEffect } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import toast from 'react-hot-toast'
+import * as z from 'zod'
+
+import { useParams } from 'common'
+import { useCreateThirdPartyAuthIntegrationMutation } from 'data/third-party-auth/integration-create-mutation'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogSection,
+  DialogTitle,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Form_Shadcn_,
+  Input_Shadcn_,
+  Separator,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+import { AwsRegionSelector } from './AwsRegionSelector'
+
+interface CreateAwsCognitoAuthIntegrationProps {
+  visible: boolean
+  onClose: () => void
+  // TODO: Remove this if this Dialog is only used for creating.
+  onDelete: () => void
+}
+
+const FORM_ID = 'create-aws-cognito-auth-integration-form'
+
+const FormSchema = z.object({
+  enabled: z.boolean(),
+  awsCognitoUserPoolId: z
+    .string()
+    .trim()
+    .min(1)
+    .regex(/^[A-Za-z0-9-_]+$/, 'The project ID contains invalid characters.'), // Only allow alphanumeric characters and hyphens.
+  awsRegion: z.string(),
+})
+
+export const CreateAwsCognitoAuthIntegrationDialog = ({
+  visible,
+  onClose,
+  onDelete,
+}: CreateAwsCognitoAuthIntegrationProps) => {
+  // TODO: Remove this if this Dialog is only used for creating.
+  const isCreating = true
+
+  const { ref: projectRef } = useParams()
+  const { mutate: createAuthIntegration, isLoading } = useCreateThirdPartyAuthIntegrationMutation({
+    onSuccess: () => {
```

Comment:
Running into an error creating a cognito auth integration - is it to do with my input?
![image](https://github.com/user-attachments/assets/14e3b636-f2c3-45d7-9d35-f0cfb867b559)

Reply:
^ feedback on API end - needs clearer error messages, i wouldnt know how to follow up here

Reply:
just FYI i also run into this error when i try to create, for example, 2 auth0 integrations immediately one after the other - again will be ideal that the API returns clearer errors

Reply:
No inline error messages in the forms >